### PR TITLE
docs: Add description for environment-specific files

### PR DESCRIPTION
**Pull request by zakuro9715**
files like `_nix.c.v` are used often but it is undocumented. So I added description.

### CODE REVIEW

File: doc/docs.md [hunk 1]
```diff
@@ -2954,6 +2956,37 @@ Full list of builtin options:
 | `gnu`, `hpux`, `haiku`, `qnx` | `cplusplus`       | `big_endian`          | |
 | `solaris`, `linux_or_macos`   | | | |
 
+### Environment specific files
+
+If a file has environment-specific suffix, it will only be compiled that environment
```
Comment (JalonSolov) [inline doc/docs.md:2961]: "If a file has an environment-specific suffix, it will only be compiled for that environment."

**Comment by spaceface777** [comment]
A `_default.c.v` suffix was also implemented recently- it would be helpful to document that as well :)

**Comment by JalonSolov** [comment]
Ah, I saw the talk, didn't know it was already there.

Probably need to beef it up a little with a more complete description of which files get used when... such as when & how `_default.c.v` will be used in addition to/in place of any other `.c.v` file.  However, that doesn't necessarily have to be in this PR.

**Comment by spytheman** [comment]
It is the opposite actually - the _default.c.v file is used by default, and will *not* be used, when a more specific platform file is found.
I should have documented it, sorry.

**Pull request merged**